src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,300 @@
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found CreateFleet parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            response = create_fleet(self._region, self._boto3_config, launch_params)
+            for error in response.get("Errors", []):
+                logger.error("Error in CreateFleet request: %s - %s", error.get("ErrorCode"), error.get("ErrorMessage"))
+
+            logger.debug("CreateFleet response: %s", response)
+            instance_ids = [inst_id for instances in response["Instances"] for inst_id in instances["InstanceIds"]]
+            return {"Instances": self._get_instances_info(instance_ids)}
+        except ClientError as e:
+            logger.error("Failed CreateFleet request: %s", e.response.get("ResponseMetadata").get("RequestId"))
+            raise e
+
+    def _get_instances_info(self, instance_ids, retry=3):
+        """
+        Describe instances to retrieve info not available from create-fleet response.
+
+        :raises EC2InstancesInfoUnavailable if after N retries it's not possible to retrieve info for all the instances
+        :raises ClientError in case of boto3 failure
+        """
+        # Wait for instances to be available in EC2
+        time.sleep(5)
+
+        # Retrieve instance info from Instance Ids
+        instances = []
+        if instance_ids:
+            logger.info(f"Retrieving instance info for instances: {instance_ids}")
+            ec2_client = boto3.client("ec2", region_name=self._region, config=self._boto3_config)
+            paginator = ec2_client.get_paginator("describe_instances")
+            response_iterator = paginator.paginate(InstanceIds=instance_ids)
+            filtered_iterator = response_iterator.search("Reservations[].Instances[]")
+            instances = [instance_info for instance_info in filtered_iterator]
+            logging.debug("Launched instances: %s", instances)
+
+            partial_info_instances = []
+            for instance_info in instances:
+                try:
+                    # Try to build EC2Instance objects retrieving all the required fields
+                    EC2Instance.from_info(instance_info)
+                except KeyError:
+                    failed_instance_id = instance_info["InstanceId"]
+                    partial_info_instances.append(failed_instance_id)
+                    instances.remove(instance_info)
```

Comment:
you don't want to modify the collection while iterating over it, make a copy if you need to do so. Also a set is more efficient than a list in this case

Reply:
You're right for the modification of the collection. 
For the set I was using a list because describe_instance_types requires a list at the end.